lib/Test/MockTime/HiRes.pm

```diff
@@ -80,7 +80,7 @@ sub gettimeofday() {
     if (defined $Test::MockTime::fixed) {
         return wantarray ? do {
             my $int_part = int($Test::MockTime::fixed);
-            ($int_part, 1_000_000 * ($Test::MockTime::fixed - $int_part))
+            ($int_part, 1_000_000 * sprintf('%.6f', ($Test::MockTime::fixed - $int_part)))
```

Comment:
>  I guess this doesn't happen for a floating point number represented in a string but I'm not sure. So please clarify this.

Maybe we should bet that the guess is correct. Let's fix it when we lose the bet.